src/main/java/io/dockstore/provision/DOSPlugin.java

```diff
@@ -49,24 +52,58 @@ public void stop() {
     @Extension
     public static class DOSPreProvision implements PreProvisionInterface {
 
-        DOSPluginUtil dosPluginUtil = new DOSPluginUtil();
         static final Set<String> SCHEME = new HashSet<>(Lists.newArrayList("dos"));
+        static final String SCHEME_PREFERENCE = "scheme-preference";
+        static final int GET_SCHEME = 0;
+        private Map<String, String> config;
+
+        DOSPluginUtil dosPluginUtil = new DOSPluginUtil();
+        List<String> preferredSchemes = new ArrayList<>();
+
+        @Override
+        public void setConfiguration(Map<String, String> map) {
+            this.config = map;
+
+            if (this.config.containsKey(SCHEME_PREFERENCE)) {
+                this.preferredSchemes = Arrays.asList(this.config.get(SCHEME_PREFERENCE).trim().split(",\\s"));
+            }
+        }
 
         public Set<String> schemesHandled() {
             return SCHEME;
         }
 
         public List<String> prepareDownload(String targetPath) {
             List<String> urlList = new ArrayList<>();
+            Map<String, List<String>> urlMap = new LinkedHashMap<>();
```

Comment:
Please add a comment describing with this map is mapping (I've gotten that suggestion many times on my own PRs). :)